src/services/security/modules/RedefineModule/index.ts

```diff
@@ -106,25 +109,36 @@ export type RedefineResponse = {
 }
 
 export class RedefineModule implements SecurityModule<RedefineModuleRequest, RedefineModuleResponse> {
+  static prepareMessage(request: RedefineModuleRequest): string {
+    const { data, safeAddress, chainId } = request
+    if (isObjectEIP712TypedData(data)) {
+      const normalizedMsg = normalizeTypedData(data)
+      return JSON.stringify(normalizedMsg)
+    } else {
+      return JSON.stringify(
+        generateTypedData({
+          safeAddress,
+          safeVersion: '1.3.0', // TODO: pass to module, taking into account that lower Safe versions don't have chainId in payload
+          chainId,
+          safeTransactionData: (data as SafeTransaction).data,
```

Comment:
Strange that we have to cast here even though `isObjectEIP712TypedData` is a type guard. I would expect that in the else case it infers to `SafeTransaction`

Reply:
I was also disappointed by this. We could have a second typeguard here of course but both is not ideal.

Reply:
I realized why... `isObjectEIP712TypedData` returns a `boolean` and not a typeguard. I rewrote it in our project as a proper typeguard.